packages/name-controller/src/NameController.ts

```diff
@@ -14,6 +14,17 @@ import type {
 import { NameType } from './types';
 
 export const FALLBACK_VARIATION = '*';
+
+/**
+ * Enumerates the possible origins responsible for setting a petname.
+ */
+export enum ORIGIN {
+  CONTROLLER = 'controller',
```

Comment:
On second thought, is `api` a little more meaningful rather than `controller` which ultimately all names will go through?

Reply:
Matt, What does "api" mean as a origin? For example, is it "The petname was created using the NameController.setName api?"

Reply:
I'm okay with moving forward using "API" where we don't specify an origin. However, I'm finding this default Origin a bit challenging to understand: It feels somewhat arbitrary and obscure to me. Could you help me understand your decision?

Reply:
Sorry Derek, I think I see the confusion, we want to default the value in any requests that set the name, but if the name is null, they logically we should set it to `null`, and similarly any new entries without a name should also have `null` since it's the origin of the name specifically, not the proposed names for example.

Reply:
Done.